internal/pkg/service/stream/storage/level/local/encoding/encoder/csv/fastcsv/writer.go

```diff
@@ -50,38 +57,46 @@ func (w *writer) WriteRow(cols *[]any) error {
 			}
 
 			// Copy verbatim everything before the special character.
-			if _, err := w.row.Write(toWrite[:stop]); err != nil {
-				return err
+			if b, err := w.row.Write(toWrite[:stop]); err != nil {
+				return n, err
+			} else {
+				n += b
 			}
 
 			toWrite = toWrite[stop:]
 
 			// Encode the special character
 			if len(toWrite) > 0 && toWrite[0] == '"' {
-				if _, err := w.row.WriteString(`""`); err != nil {
-					return err
+				if b, err := w.row.WriteString(`""`); err != nil {
+					return n, err
```

Comment:
Do we care about `n` when we have `error` while writing? E.g I wrote 1MB of characters but end up with error. The meter does not matter because there will be issue that it cannot fit the `row` anymore.

Reply:
On error, the buffer is discarded, either the entire row is written or nothing, so it should return `0`.

Reply:
So basically it would be better to return `0, err`.